### PR TITLE
Add Tracing Integration Test

### PR DESCRIPTION
**Pull request by bbcadela**

### CODE REVIEW

File: grpc/server/pom.xml [hunk 1]
```diff
@@ -125,6 +130,21 @@
...
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>io.zipkin.zipkin2</groupId>
+            <artifactId>zipkin-junit</artifactId>
+            <version>${version.zipkin-junit}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.vintage</groupId>
```
Comment (thegridman) [inline grpc/server/pom.xml]: Why did you need this dependency

File: grpc/server/pom.xml [hunk 2]
```diff
@@ -125,6 +130,21 @@
             <artifactId>javax.activation-api</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (thegridman) [inline grpc/server/pom.xml:133]: This dependency needs to be copied to the top level pom.xml and then the <version> tag removed from this pom.xml

File: grpc/server/pom.xml [hunk 3]
```diff
@@ -31,6 +31,11 @@
     <artifactId>helidon-grpc-server</artifactId>
     <name>Helidon gRPC Server</name>
 
+    <properties>
+        <version.zipkin-junit>2.12.5</version.zipkin-junit>
```
Comment (thegridman) [inline grpc/server/pom.xml:35]: This property should be in the top level pom.xml

File: grpc/server/src/test/java/io/helidon/grpc/server/TracingIT.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+    // ----- test methods ---------------------------------------------------
+
+    @Test
+    public void shouldTraceMethodNameAndHeaders() throws Exception{
+        // call the gRPC Echo service so that there should be tracing span sent to zipkin server
+        EchoServiceGrpc.newBlockingStub(channel).echo(Echo.EchoRequest.newBuilder().setMessage("foo").build());
+
+        // wait a bit for span from zipkin
+        Thread.sleep(2000);
```
Comment (thegridman) [inline grpc/server/src/test/java/io/helidon/grpc/server/TracingIT.java]: We can probably introduce Bedrock end use `Eventually.assertThat()` instead of sleeping. We'll have to see then whether the Helidon team object to us using Bedrock. You will need to add a dependency on groupId `com.oracle.bedrock` artifactId `bedrock-testing-support` version `5.0.11` and put it into the top level pom.xml and then into this project's pom.xml without the version element.

As Bedrock creates a proxy (mock) of the class used in `Eventually.assertThat()` we cannot use the `zipkin` value directly as the `ZipkinRule` class is final so it cannot be proxied. So, you need to do this in a bit of a round-about way. First create a method in this class like this:
```
public int getSpanCount() {
    return zipkin.collectorMetrics().spans();
}
```
Then instead of the sleep you can do:
```
Eventually.assertThat(invoking(this).getSpanCount(), is(1));
```
Now inside `invoking` we are creating a proxy of this test class, which will work fine. This will then assert that eventually there is at least one span then proceed with the rest of your test.

**Review by thegridman — CHANGES_REQUESTED** [review]

File: grpc/server/pom.xml (hunk 3 above)
Comment (aseovic) [reply grpc/server/pom.xml:35]: Yup, I was gonna suggest the same.

See https://github.com/oracle/helidon/blob/master/DEV-GUIDELINES.md

File: grpc/server/pom.xml (hunk 2 above)
Comment (aseovic) [reply grpc/server/pom.xml:133]: I'm not sure I understand/agree here: which top-level POM?

Per https://github.com/oracle/helidon/blob/master/DEV-GUIDELINES.md, we should add every new dependency to `dependencyManagement` section of the *root* POM (I'm assuming that's what JK means by "top-level"), but then we still need to specify it (without version, of course) in the POM that uses it.

For now, that is the server POM, so I believe it should stay here. If we end up needing it in the client POM as well, which I suspect we will, we can decide whether we want to move it one level up, to the main `grpc` POM, or simply add it to the client POM as well.

Personally, I feel the latter is better, as it documents dependency clearly in the project that actually uses it, but I could be persuaded to move it up if there is a good reason for it.

File: grpc/server/src/test/java/io/helidon/grpc/server/TracingIT.java (hunk 4 above)
Comment (aseovic) [reply grpc/server/src/test/java/io/helidon/grpc/server/TracingIT.java]: Yeah, we definitely shouldn't sleep... We know that random sleeps caused all kinds of issues in Coherence tests and that we went through hell to get rid of them, so we shouldn't even start with them here.

I think Bedrock is fine, and Helidon guys won't object to it, especially if the WLS team is starting to use it as well, but we should definitely discuss it with them next time we meet.

**Review by aseovic — CHANGES_REQUESTED** [review]

**Review by thegridman — APPROVED** [review]
This looks ok now.
